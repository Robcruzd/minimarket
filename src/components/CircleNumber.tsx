import { StyleSheet, Text } from "react-native"

const CircleNumber = (props: {
    style?: {}; 
    quantity: number
}) => {
    return (
        <Text testID="circleNumber" style={[styles.textStyle, props.style]}>{props.quantity}</Text>
    )
}

export default CircleNumber;

const styles = StyleSheet.create({
    textStyle : {
        textAlign:"center",
        textAlignVertical:"center",
        zIndex: 2,
        backgroundColor:"#A51BB6", 
        color: "#fff",
        position:"absolute",
        width:30,
        height:30,
        borderRadius:50,
        top:5,
        left:5
        }
})