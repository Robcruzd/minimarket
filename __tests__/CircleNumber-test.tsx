import React from 'react';
import { create } from 'react-test-renderer';
import CircleNumber from '../src/components/CircleNumber';

describe('<CircleNumber />', () => {
  test('renders correctly with props', () => {
    const tree = create(<CircleNumber quantity={5} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('renders correct quantity', () => {
    const quantity = 5;
    const circleRenderer = create(<CircleNumber quantity={quantity} />);
    const circleInstance = circleRenderer.root;
    expect(circleInstance.findByProps({testID: 'circleNumber'}).props.children).toBe(quantity);
  });

  test('renders with custom style', () => {
    const customStyle = { color: '#00FF00' };
    const circleRenderer = create(<CircleNumber quantity={5} style={customStyle} />);
    const circleInstance = circleRenderer.root;
    expect(circleInstance.findByProps({testID: 'circleNumber'}).props.style).toContain(customStyle);
  });
});