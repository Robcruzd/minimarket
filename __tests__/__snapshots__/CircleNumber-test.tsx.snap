// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleNumber /> renders correctly with props 1`] = `
<Text
  style={
    [
      {
        "backgroundColor": "#A51BB6",
        "borderRadius": 50,
        "color": "#fff",
        "height": 30,
        "left": 5,
        "position": "absolute",
        "textAlign": "center",
        "textAlignVertical": "center",
        "top": 5,
        "width": 30,
        "zIndex": 2,
      },
      undefined,
    ]
  }
  testID="circleNumber"
>
  5
</Text>
`;
